pyquil/tests/test_api.py

```diff
@@ -60,156 +67,158 @@
 RB_REPLY = [Program("H 0\nH 0\n"), Program("PHASE(pi/2) 0\nPHASE(pi/2) 0\n")]
 
 
-def test_sync_run_mock(qvm: QVMConnection, httpx_mock: HTTPXMock):
-    mock_qvm = qvm
-    mock_endpoint = mock_qvm.client.qvm_url
-    httpx_mock.add_response(
-        method="POST",
-        url=mock_endpoint,
-        match_content=json.dumps(
-            {
-                "type": "multishot",
-                "addresses": {"ro": [0, 1]},
-                "trials": 2,
-                "compiled-quil": "DECLARE ro BIT[2]\nH 0\nCNOT 0 1\nMEASURE 0 ro[0]" + "\nMEASURE 1 ro[1]\n",
-                "rng-seed": 52,
-            }
-        ).encode(),
-        json={"ro": [[0, 0], [1, 1]]},
-    )
-
-    assert mock_qvm.run(BELL_STATE_MEASURE, [0, 1], trials=2) == [[0, 0], [1, 1]]
+@mock.patch("pyquil.api.client._client_provider.QVMClient", autospec=True)
```

Comment:
We can now properly mock the appropriate sub-client, rather than skipping a layer and mocking the http (since the http mocking is already covered in the client tests)